packages/web-react/src/types/textField.ts

```diff
@@ -1,23 +1,16 @@
-import { ChildrenProps, StyleProps, TextInputProps, TransferProps, HelperTextProps } from './shared';
-import { LabelProps } from './label';
-import { MessageProps } from './message';
-import { PasswordToggleAdornmentProp } from './shared/adornments';
+import { InputCommonProps, InputBaseProps, PasswordToggleAdornmentProp, SpiritInputElementProps } from './shared';
 
 export type TextFieldType = 'email' | 'number' | 'password' | 'search' | 'tel' | 'text' | 'url';
 
+export type TextFieldElementBaseProps = Omit<SpiritInputElementProps, 'type' | 'required'>;
+
 export interface TextFieldProps
-  extends ChildrenProps,
-    StyleProps,
-    LabelProps,
-    MessageProps,
-    HelperTextProps,
-    TextInputProps,
-    TransferProps,
+  extends TextFieldElementBaseProps,
+    InputCommonProps,
```

Comment:
I do not think that using both `Common` and `Base` is a good idea. Both are interchangeable and both with very similar meanings. And also I think it should be up to the component to fabricate the final type.

Reply:
fixed